### PR TITLE
ZCS-5109 Fixing loading of config from LDAP

### PR DESCRIPTION
**Pull request by rupalid**
Added code to load configuration at domain level and if not available then at config level.
Will send a separate PR for the unit test.

Test and verified that contacts for google, yahoo and facebook are synced.

### CODE REVIEW

File: src/java/com/zimbra/oauth/handlers/impl/FacebookOAuth2Handler.java [hunk 1]
```diff
@@ -362,11 +365,14 @@ protected String getPrimaryEmail(JsonNode credentials) throws ServiceException {
      * @return The Facebook App token
      * @throws ServiceException If there was an issue with the request
      */
-    protected String getAppToken() throws ServiceException {
+    protected String getAppToken(Account account) throws ServiceException {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/handlers/impl/FacebookOAuth2Handler.java]: Add new param to docblock.

File: src/java/com/zimbra/oauth/handlers/impl/FacebookOAuth2Handler.java [hunk 2]
```diff
@@ -319,13 +322,13 @@ protected String inErrorCodeRange(String errorCode) {
      *             address
      */
     @Override
-    protected String getPrimaryEmail(JsonNode credentials) throws ServiceException {
+    protected String getPrimaryEmail(JsonNode credentials, Account acct) throws ServiceException {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/handlers/impl/FacebookOAuth2Handler.java]: Add new param to docblock.

File: src/java/com/zimbra/oauth/resources/ZOAuth2Servlet.java [hunk 3]
```diff
@@ -62,14 +71,22 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
         final String client = pathParams.get("client");
         String location = OAuth2Constants.DEFAULT_SUCCESS_REDIRECT;
 
+        String encodeAuthToken = getEncodedAuthTokenFromCookie(req);
+        Account account = getAccount(req, encodeAuthToken);
+        ZimbraLog.extensions.info("Account is:%s", account);
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/resources/ZOAuth2Servlet.java]: Do we want to make this debug level?

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 4]
```diff
@@ -44,87 +47,81 @@ protected LdapConfiguration(String appName) {
      * @param key A key to lookup
      * @return A value for a given key
      */
-    @Override
-    public String getString(String key) {
+
+    public  String getString(String key, String appName, Account account) {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Add new params to docblock.

File: src/java/com/zimbra/oauth/utilities/LdapConfiguration.java [hunk 5]
```diff
@@ -44,87 +47,81 @@ protected LdapConfiguration(String appName) {
...
+        return getConfig(key, appName, account);
     }
 
     /**
      * @param oauth related config key
      * @param appName client app name
      * @return value for the associated key
      */
-    public String getConfig(String key, String appName) {
+    public String getConfig(String key, String appName, Account acct) {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/LdapConfiguration.java]: Add new param to docblock.

File: src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java [hunk 6]
```diff
@@ -51,10 +52,10 @@
      * @return Location to redirect to
      * @throws ServiceException If there are issues
      */
-    public static final String authorize(String client, String relay) throws ServiceException {
+    public static final String authorize(String client, String relay, Account acct) throws ServiceException {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java]: Add new param to docblock.

File: src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java [hunk 7]
```diff
@@ -68,7 +69,7 @@ public static final String authorize(String client, String relay) throws Service
      * @throws ServiceException If there are issues
      */
     public static String authenticate(String client, Map<String, String[]> queryParams,
-        String zmAuthToken) throws ServiceException {
+        Account account, String zmAuthToken) throws ServiceException {
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java]: Add new param to docblock.

Comment (rupalid) [reply src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java]: Done

**Pull request merged**